cmd/agent/app/server.go

```diff
@@ -49,28 +49,32 @@ func NewAgentCommand(a *Agent, o *options.GrpcProxyAgentOptions) *cobra.Command
 		Use:  "agent",
 		Long: `A gRPC agent, Connects to the proxy and then allows traffic to be forwarded to it.`,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return a.run(o)
+			stopCh := make(chan struct{})
```

Comment:
I moved it to here to make `stopCh` an argument to `Run(...)`, so that tests can call the `Run` method and shutdown at the end. How does moving it to `main()` improve graceful shutdown? Actually, it looks like the server command explicitly installs a signal handler for graceful shutdown. Would it be better to just reuse that code here?

Reply:
I started to make the signal handler change, but I'd prefer to make it a separate PR to derisk this PR and keep the non-test changes minimal.

Reply:
I agree with separate PR, especially since this commit is so large.

(The server signal handler pattern seems applicable to agent. But I think we should be cautious with making changes.)